source/server/filter_chain_manager_impl.cc

```diff
@@ -465,5 +568,36 @@ void FilterChainManagerImpl::convertIPsToTries() {
     destination_ips_pair.second = std::make_unique<DestinationIPsTrie>(destination_ips_list, true);
   }
 }
+
+std::unique_ptr<FilterChainFactoryContextCallback>
+FilterChainManagerImpl::createFilterChainFactoryContextCallback(
+    Configuration::FactoryContext& parent_context) {
+  return std::make_unique<FilterChainContextCallbackImpl>(*this, parent_context);
+}
+
+FilterChainManagerImpl::FilterChainContextCallbackImpl::FilterChainContextCallbackImpl(
+    FilterChainManagerImpl& parent, Configuration::FactoryContext& parent_context)
+    : parent_(parent), parent_context_(parent_context) {}
+
+void FilterChainManagerImpl::FilterChainContextCallbackImpl::prepareFilterChainFactoryContexts() {
+  // The previous add might fail. Clean it here.
+  parent_.factory_contexts_.clear();
```

Comment:
Expand? How does this happen? Do we need this behavior in this PR?

Reply:
It's not too bad with this PR. Please... it will be used in next PR and I didn't pollute the `include/`

Reply:
If you want to keep it in this PR then please add a bunch more comments. I don't understand the batching behavior and why it's needed. Once you add more comments I will take another pass. Thank you!

Reply:
update. Removed batch related functions